shared/src/main/scala/scala/xml/Utility.scala

```diff
@@ -51,13 +51,9 @@ object Utility extends AnyRef with parsing.TokenTests {
   }
 
   private def combineAdjacentTextNodes(children: Node*): Seq[Node] = {
-    children.foldLeft(Seq.empty[Node]) { (acc, n) =>
-      (acc.lastOption, n) match {
-        case (Some(Text(l)), Text(r)) => {
-          acc.dropRight(1) :+ Text(l + r)
-        }
-        case _ => acc :+ n
-      }
+    children.foldRight(Seq.empty[Node]) {
+      case (Text(left), Text(right) +: accMinusLast) => Text(left + right) +: accMinusLast
+      case (n, acc) => n +: acc 
     }
```

Comment:
> I wasn't aware you can pattern match the last element in a list

It is pattern matching the front of the list with `foldRight` ("fold right starts from the **left**" is how I remember it).  Is it important to start from the end?  I presume it's not.  You're just trying to merge adjacent `Text`s.

Reply:
Ah, ok.  Yeah, it's a lot for a person to keep in their head... between the colons, plus signs and foldings.